testsuite/MDAnalysisTests/topology/test_minimal.py

```diff
@@ -0,0 +1,109 @@
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import absolute_import
+
+import itertools
+import numpy as np
+import pytest
+
+import MDAnalysis as mda
+from MDAnalysis.topology.MinimalParser import MinimalParser
+
+from MDAnalysisTests.datafiles import (
+    DCD,
+    INPCRD,
+    LAMMPSdcd2,
+    NCDF,
+    TRJ,
+    TRJncdf,
+    TRR,
+    TRZ,
+    XTC,
+)
+
+
+working_readers = pytest.mark.parametrize(
+    'filename,expected_n_atoms', [
+        (DCD, 3341),
+        (INPCRD, 5),
+        (LAMMPSdcd2, 12421),
+        (NCDF, 2661),
+        (TRR, 47681),
+        (XTC, 47681),
+        (np.zeros((1, 10, 3)), 10),  # memory reader default
+    ])
+
+@working_readers
+def test_minimal_parser(filename, expected_n_atoms):
+    with MinimalParser(filename) as p:
+        top = p.parse()
+    assert top.n_atoms == expected_n_atoms
+
+
+@pytest.mark.parametrize('filename', [
+    TRJ,
+    TRJncdf,
+    TRZ,
+])
+def test_minimal_parser_fail(filename):
+    with MinimalParser(filename) as p:
+        with pytest.raises(NotImplementedError):
+            p.parse()
+
+
+@working_readers
+def test_universe_with_minimal(filename, expected_n_atoms):
+    u = mda.Universe(filename)
+
+    assert len(u.atoms) == expected_n_atoms
+
+
+def memory_possibilities():
+    # iterate over all possible shapes for a MemoryReader array
+    # number of frames, atoms and coordinates
+    n = {'f': 1, 'a': 10, 'c': 3}
+    for permutation in itertools.permutations('fac', 3):
+        order = ''.join(permutation)
+        array = np.zeros([n[val] for val in permutation])
+
+        yield array, order
+
+
+memory_reader = pytest.mark.parametrize('array,order', list(memory_possibilities()))
+
+@memory_reader
+def test_memory_minimal_parser(array, order):
+    with MinimalParser(array) as p:
+        top = p.parse(order=order)
+    assert top.n_atoms == 10
+
+@memory_reader
+def test_memory_universe(array, order):
+    u = mda.Universe(array, order=order)
+
+    assert len(u.atoms) == 10
+
+
+def test_minimal_n_atoms_kwarg():
```

Comment:
Could parametrize to the same set of inputs as `test_minimal_parser_fail` above maybe? If I understand correctly those formats can only be "minimally handled" if the number of atoms is provided in advance?

Reply:
yep good point, done